source/common/stream_info/filter_state_impl.cc

```diff
@@ -79,7 +79,7 @@ FilterStateImpl::getDataSharedMutableGeneric(absl::string_view data_name) {
 
   FilterStateImpl::FilterObject* current = it->second.get();
   if (current->state_type_ == FilterState::StateType::ReadOnly) {
-    throw EnvoyException("FilterState tried to access immutable data as mutable.");
+    IS_ENVOY_BUG("FilterState tried to access immutable data as mutable.");
```

Comment:
Can you add a filter key here and elsewhere? It would be easier to scavenge the logs.